### ISSUE TITLE
[feature request] add highlight to the yanked region

### ISSUE THREAD

**Issue by aleksey-rowan** [replit/codemirror-vim#121]
Please add an option to highlight the yanked region like in https://github.com/machakann/vim-highlightedyank or natively in Neovim. Thanks!

**Comment by nightwing** [comment]
@sergeichestakov should we add this feature, or  should we limit the scope to vim defaults only?

**Comment by sergeichestakov** [comment]
yeah I think we'll want to stick with defaults as close as possible. we can possibly add it as a config option tho

**Comment by aleksey-rowan** [comment]
> yeah I think we'll want to stick with defaults as close as possible. we can possibly add it as a config option tho

I understand the importance to sticking with the defaults and avoiding scope creep, but if this can be added an an option, that would be amazing. Highlighting yanks is one of those quality-of-life features--I believe--that almost everyone adds to their configs right away.

Thank you for considering this anyway!

**Comment by joelostblom** [comment]
This is actually [available as a default in Neovim](https://neovim.io/doc/user/lua.html#vim.highlight), and it is super helpful. Especially  for confirming the yanked region (especially when grabbing a paragraph or targeting a character where it is easy to overlook that there is another of the same character earlier or grab an extra whitespace), and also for teaching/presenting so that others can easily follow along what you are doing. I raised this in the jupyterlab-vim repo originally https://github.com/jupyterlab-contrib/jupyterlab-vim/issues/117 and posted this example:

![](https://camo.githubusercontent.com/3fc438b068ede98cf3197fa402e4cd2a669eb8f0c7448f24406870396cd06eaa/687474703a2f2f692e696d6775722e636f6d2f48756c795a366e2e676966)

**Comment by aleksey-rowan** [comment]
Shameless plug here. I made an Obsidian plugin to do just that--highlight the yanked region. If you are using Obsidian and were missing that feature, check it out: 

https://github.com/aleksey-rowan/obsidian-vim-yank-highlight

![](https://raw.githubusercontent.com/aleksey-rowan/obsidian-vim-yank-highlight/master/assets/demo.gif)

**Comment by joelostblom** [comment]
That's great, thank you @aleksey-rowan ! Is there anything from your approach that would be useful/able to be transferred over if this feature is approved to be added in codemirror directly? Or is it completely different to implement it in obsidian?

**Comment by aleksey-rowan** [comment]
@joelostblom, there is almost nothing Obsidian-specific apart from the plugin wrapper, so I think the same approach can be added to the codemirror-vim directly. It should be even easier since you can hook into the pushTest/yank handler instead of listening to key events and then trying to find yanked text in the viewport. 

There is [another Obsidian yank plugin](https://github.com/kometenstaub/obsidian-vim-yank-highlight/) that does a similar thing by monkey-patching codemirror-vim, so their approach would be even more adaptable to be bundled in.
